tools/annotato/src/annotator_app.rs

```diff
@@ -0,0 +1,278 @@
+
+        let model_annotations = ModelAnnotations::try_new(annotation_json_path)?;
+
+        let paths = image_paths
+            .into_iter()
+            .map(|image_path| {
+                let label_path = Self::convert_image_to_label_path(&image_path)?;
+                Ok(Paths::new(image_path, label_path))
+            })
+            .collect::<Result<VecDeque<_>>>()
+            .expect("failed to build paths");
+
+        let phase = if skip_introduction {
+            AnnotationPhase::Labelling { current_index: 0 }
+        } else {
+            AnnotationPhase::Started
+        };
+
+        Ok(AnnotatorApp {
+            phase,
+            paths,
+            label_widget: LabelWidget::default(),
+            model_annotations,
+        })
+    }
+
+    fn next(&mut self) {
+        self.phase = match self.phase {
+            AnnotationPhase::Started => AnnotationPhase::Labelling { current_index: 0 },
+            AnnotationPhase::Labelling { current_index } => {
+                let new_index = current_index + 1;
+                if new_index < self.paths.len() - 1 {
+                    AnnotationPhase::Labelling {
+                        current_index: new_index,
+                    }
+                } else {
+                    AnnotationPhase::Finished
+                }
+            }
+            AnnotationPhase::Finished => AnnotationPhase::Finished,
+        };
+    }
+
+    fn previous(&mut self) {
+        self.phase = match self.phase {
+            AnnotationPhase::Started => AnnotationPhase::Started,
+            AnnotationPhase::Labelling { current_index } => {
+                if current_index > 1 {
+                    let new_index = current_index - 1;
+                    AnnotationPhase::Labelling {
+                        current_index: new_index,
+                    }
+                } else {
+                    AnnotationPhase::Started
+                }
+            }
+            AnnotationPhase::Finished => {
+                let new_index = self.paths.len() - 1;
+                AnnotationPhase::Labelling {
+                    current_index: new_index,
+                }
+            }
+        };
+    }
+
+    fn load_image(&mut self) -> Result<()> {
+        let index = match self.phase {
+            AnnotationPhase::Started => 0,
+            AnnotationPhase::Labelling { current_index } => current_index,
+            AnnotationPhase::Finished => self.paths.len() - 1,
+        };
+
+        if let Some(paths) = self.paths.get_mut(index) {
+            if self.label_widget.has_paths(paths) {
+                return Ok(());
+            }
+            let annotations = self
+                .model_annotations
+                .for_image(
+                    &paths
+                        .image_path
+                        .file_name()
+                        .unwrap()
+                        .to_str()
+                        .unwrap()
+                        .to_string(),
```

Comment:
uff... is there no shorter way for that?

Reply:
I could not find any other way :(